code-challenges.js

```diff
@@ -12,64 +13,140 @@ var num2 = 0
+}
 
-
-
+console.log(onlyNumbersSort(mixedDataArray), "\n");
 
 // --------------------4) Create a function that takes in a string and logs the index of the first vowel.
+console.log("4:");
 
 var vowelTester1 = "learn"
 // Expected output: 1
 var vowelTester2 = "throw"
 // Expected output: 3
 
-
-
-
+const firstVowelIndex = (string) => {
+    let word = string.toLowerCase();
+    let vowelArr = [];
+    // deconstruct string into array of letters
+    let  arrLetters = word.split("");
+    // map through array checking for vowels, if found, push index to vowelArr
+    arrLetters.map((letter, index) => {
+        if(letter === "a" || letter === "e" || letter === "i" || letter === "o" || letter === "u" || letter === "y") {
+            vowelArr.push(index);
+        }
+    })
+    // return first value in vowelIndex
+    return vowelArr[0];
+}
+
+console.log(firstVowelIndex(vowelTester1));
+console.log(firstVowelIndex(vowelTester2), "\n");
 
 
 // --------------------5) Create a function that takes three arguments - two numbers and a mathematical operation (+, -, *, /) and performs the given calculation. If the input tries to divide by 0, return: "Can't divide by 0!"
-
-
-
-
+console.log("5:");
+
+const calculator = (num1, operator, num2) => {
+    // Type check for numbers
+    if(typeof num1 === "number" && typeof num2 === "number"){
+    // If else statements for each operator 
+        if(operator === "+"){
+            return num1 + num2;
+        } else if(operator === "-"){
+            return num1 - num2;
+        } else if(operator === "*"){
+            return num1 * num2;
+        } else if(operator === "/"){
+            // if statement accounting for 0's
+            if(num2 === 0){
+                return "Can't divide by 0!";
+            } else {
+                return num1 / num2;
+            }
+        } else {
+            return "Please input two numbers and a mathematical operator"
+        }
+    } else {
+        return "Please input two numbers and a mathematical operator"
+    }
+}
 
 // Uncomment and use the following console logs to test your function
-// console.log(calculator(3, "*", 9))
+console.log(calculator(3, "*", 9))
 // Expected output: 27
 
-// console.log(calculator(16, "+", 3))
+console.log(calculator(16, "+", 3))
 // Expected output: 19
 
-// console.log(calculator(89, "/", 0))
+console.log(calculator(89, "/", 0), "\n")
 // Expected output: "Can't divide by 0!"
 
-
-
 // --------------------6) Create a function that takes in the following variable and returns only the websites that end in ".io".
+console.log("6:");
 
 var websites = ["codepen.io", "codecademy.com", "coursera.org", "codepen.io", "udemy.com", "pluralsight.com", "udacity.com", "sitepoint.com"]
 // Expected output: "codepen.io" "codepen.io"
 
+const onlyDotIos = (array) => {
+    // Filter array for only the strings with ".io"
+    let filteredArray = array.filter(site => site.includes(".io") === true);
+    return filteredArray;
```

Comment:
Clean! Another way we can make it even more clean, similar to an above example I provided: 

`return array.filter(site => site.includes(".io") === true);`

Since we are already passing a variable into the function, we can take advantage of that by not needing to create ANOTHER variable that is basically just declared to be the one passed into it. Since we are just going to return the same variable right after, we can skip the additional return by just returning the original variable/array right off the bat. 

Again, nothing wrong at all with how you wrote it, but just something I want to bring up for you to look into!